### PR TITLE
docs: Add documentation for dataflow config

### PR DESCRIPTION
**Pull request by Deep1998**
Add documentation for dataflow config.

Jekyll link: https://deep1998.github.io/harbourbridge/ui/prepare-migration/dataflow-tuning.html

### CODE REVIEW

File: docs/ui/prepare-migration/dataflow-config.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+layout: default
+title: Configure Dataflow (Optional)
+parent: Prepare Migration Page
+grand_parent: SMT UI
+nav_order: 4
+---
+
+# Setting up connection profiles
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-config.md:9]: Change title

File: docs/ui/prepare-migration/dataflow-config.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+{: .no_toc }
+
+In case of minimal downtime migration, the dataflow jobs launched by the Spanner Migration Tool can be OPTIONALLY configured with custom runtime environment variables such as MaxWorkers, NumWorkers etc.
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-config.md:12]: Nit: Optionally as bold instead of in capitals

File: docs/ui/prepare-migration/dataflow-config.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+{: .no_toc }
+
+In case of minimal downtime migration, the dataflow jobs launched by the Spanner Migration Tool can be OPTIONALLY configured with custom runtime environment variables such as MaxWorkers, NumWorkers etc. 
+
+{: .highlight }
+The step shows configured (via a green tick) by default, denoting it is optional to configure.
+
+To configure dataflow, first specify the target database in the 'Configure Spanner Database' step. This enables the configure button for the remaining steps. The configurable fields for Dataflow are mentioned below.
+
+![](https://services.google.com/fh/files/misc/dataflow-config.png)
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-config.md:19]: The screenshot looks too small to understand the different fields in the form. Would suggest snipping and having only the form in the screenshot for clarity

File: docs/ui/prepare-migration/dataflow-config.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  <summary>
+    Table of contents
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+## Host Project Id
+Specify the project id of the VPC that you want to use. This is required in order to use private connectivity. By default, this is assumed to be the same as Spanner project.
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-config.md:31]: We need to specify this is to be used for shared VPC, and the project used here will be the host project

File: docs/ui/prepare-migration/dataflow-config.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+# Setting up connection profiles
+{: .no_toc }
+
+In case of minimal downtime migration, the dataflow jobs launched by the Spanner Migration Tool can be OPTIONALLY configured with custom runtime environment variables such as MaxWorkers, NumWorkers etc. 
+
+{: .highlight }
+The step shows configured (via a green tick) by default, denoting it is optional to configure.
+
+To configure dataflow, first specify the target database in the 'Configure Spanner Database' step. This enables the configure button for the remaining steps. The configurable fields for Dataflow are mentioned below.
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-config.md:17]: We should add the link of the Dataflow documentation which talks about networks and subnetworks.

Comment (manitgupta) [reply docs/ui/prepare-migration/dataflow-config.md:17]: We should also add reference to general Dataflow configuration parameters (which may not be part of the form) here for completeness.

File: docs/ui/prepare-migration/dataflow-config.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+Specify the name of the VPC subnetwork to use. For private connectivity, specify both the VPC network and subnetwork,
+
+## Max Workers
+Specify the max workers for the dataflow job(s). By default, set to 50.
+
+## Number of Workers
+Specify the initial number of workers for the dataflow job(s). By default, set to 1.
+
+## Service Account Email
+Specify a custom service account email to run the job as. Uses the default compute engine service account if not specified.
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-config.md:46]: Need to add link to further documentation here

File: docs/ui/prepare-migration/dataflow-config.md (hunk 4 above)
Comment (Deep1998) [reply docs/ui/prepare-migration/dataflow-config.md:31]: This needs to be specified for all cases: normal as well as shared VPCs. Added a line for shared VPC as well though.

File: docs/ui/prepare-migration/dataflow-config.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+{: .highlight }
+Currently, we only allow configuring a subset of options available in Dataflow via SMT. For the full list of options available, click [here](https://cloud.google.com/dataflow/docs/reference/pipeline-options).
+
+## Host Project Id
+Specify the project id of the VPC that you want to use. This is required in order to use private connectivity. By default, this is assumed to be the same as Spanner project.
```
Comment (bharadwaj-aditya) [inline docs/ui/prepare-migration/dataflow-config.md:35]: Let's mention what are the defaults, i.e. what happens if these fields are not provided

**Review by bharadwaj-aditya — CHANGES_REQUESTED** [review]

File: docs/ui/prepare-migration/dataflow-config.md (hunk 7 above)
Comment (Deep1998) [reply docs/ui/prepare-migration/dataflow-config.md]: Done

File: docs/ui/prepare-migration/dataflow-tuning.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+layout: default
+title: Tune Dataflow (Optional)
+parent: Prepare Migration Page
+grand_parent: SMT UI
+nav_order: 4
+---
+
+# Tune Dataflow (Optional)
+{: .no_toc }
+
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-tuning.md:11]: Let's an `{ important }` tag that the top saying "This configuration is relevant especially you want to run Dataflow inside a VPC."

File: docs/ui/prepare-migration/dataflow-tuning.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    Table of contents
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+{: .highlight }
+
+Currently, we only allow configuring a subset of options available in Dataflow via SMT. For the full list of options available, click [here](https://cloud.google.com/dataflow/docs/reference/pipeline-options).
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-tuning.md:32]: Use `Spanner migration tool` instead of "we".

Lets also rephrase this to - SMT exposes the most frequently changed dataflow configurations to the user. Please reach out to us <insert github issue ref> if you have a use-case that is not satisfied by the provided configurations.

IMO thinking the URL for full pipeline options is not helpful here since in the customer will not be use those configurations via HB, so it can be misleading.

File: docs/ui/prepare-migration/dataflow-tuning.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+### VPC Network
+Specify the name of the VPC network to use. For private connectivity, specify both the VPC network and subnetwork. If no network and subnetwork is provided, the [default](https://cloud.google.com/dataflow/docs/guides/specifying-networks#specifying_a_network_and_a_subnetwork) network is used.
+
+Present under the Networking section of the form. 
+
+### VPC Subnetwork
+Specify the name of the VPC subnetwork to use. For private connectivity, specify both the VPC network and subnetwork. If no network and subnetwork is provided, the [default](https://cloud.google.com/dataflow/docs/guides/specifying-networks#specifying_a_network_and_a_subnetwork) network is used.
+
+Present under the Networking section of the form.
```
Comment (manitgupta) [inline docs/ui/prepare-migration/dataflow-tuning.md:49]: Let's add a note here about what we do under which type of configuration as far as enabling/disabling public IPs for the Dataflow VMs are concerned.

For example - We disable public IPs when running inside a VPC. On that note, I think we should also expose enabling/disabling public IPs directly on the UI. Krishna faced this issue and I think its a valid configuration to add.

Comment (Deep1998) [reply docs/ui/prepare-migration/dataflow-tuning.md:49]: Added a note explaining this. 

As for the option to configure this, we decided to remove this based on recommendations for the dataflow team.

File: docs/ui/prepare-migration/dataflow-tuning.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+![](https://services.google.com/fh/gumdrop/preview/misc/preset-flags.png)
+
+### Dataflow Project
+Specify the project to run the dataflow job in.
+
+### Dataflow Location
+Specify the region to run the dataflow job in. It is recommended to keep the region same as Spanner region for performance. Example: us-central1
+
+### GCS Template Path
+Cloud Storage path to the template spec. Use this to run launch dataflow with custom templates. Example: `gs://my-bucket/path/to/template`
```
Comment (bharadwaj-aditya) [inline docs/ui/prepare-migration/dataflow-tuning.md:105]: It might makes sense to add a link here to the dataflow page here so that folks can read up more in detail there if required.

**Review by bharadwaj-aditya — APPROVED** [review]
